CardScan/Classes/CreditCardOcr/CreditCardOcrImplementation.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  CreditCardOcr.swift
+//  ocr-playground-ios
+//
+//  Created by Sam King on 3/19/20.
+//  Copyright © 2020 Sam King. All rights reserved.
+//
+import UIKit
+
+/**
+ Base class for any OCR prediction systems. All implementations must override `recognizeCard` and update the `frames`
+ and `computationTime` member variables
+ */
+
+class CreditCardOcrImplementation {
+    let dispatchQueue: DispatchQueue
+    var frames = 0
+    var computationTime = 0.0
+    let startTime = Date()
+    
+    var framesPerSecond: Double {
+        return Double(frames) / -startTime.timeIntervalSinceNow
+    }
+    
+    var mlFramesPerSecond: Double {
+        return Double(frames) / computationTime
+    }
+    
+    init(dispatchQueue: DispatchQueue) {
+        self.dispatchQueue = dispatchQueue
+    }
+    
+    func recognizeCard(in fullImage: CGImage, roiRectangle: CGRect) -> CreditCardOcrPrediction {
+        preconditionFailure("This method must be overridden")
+    }
+    
+    func croppedImage(fullCardImage: CGImage, roiRectangle: CGRect) -> CGImage? {
```

Comment:
would it make sense to put these cropping functions in imageUtils?

Reply:
yeah it does, but let's do that later. There is still a ton of refactoring that we need to do.